sql/item_subselect.h

```diff
@@ -1641,6 +1664,8 @@ class Subq_materialization_tracker
         return "index_lookup;array merge for partial match";
       case Strategy::PARTIAL_MATCH_SCAN:
         return "index_lookup;full scan for partial match";
+      case Strategy::PARTIAL_MATCH_SINGLE_COLUMN:
+        return "index_lookup;constant for partial match";
```

Comment:
This sounds very cryptic for user.  Internally, we handle this through `subselect_partial_match_engine` but 
imagine a user running
``` 
 col1 in (select col2 from t2)
```
and getting "partial match"?   Please consider changing to 
`check nulls; index_lookup`  or `null-aware index_lookup`  or something like that.   (It should be still different from just `index_lookup` to allow us developers to distinguish from COMPLETE_MATCH).

Reply:
... The same with `subselect_single_column_partial_engine`.  This looks puzzling. Maybe, "subselect_single_column_match_engine" ?

Reply:
Good point, I like the names you suggested. I have also renamed the strategy to `Strategy::SINGLE_COLUMN_MATCH` in a next revision.